sdk/daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/v2/ledgerinteraction/SubmitError.scala

```diff
@@ -403,6 +403,25 @@ class SubmitErrors(majorLanguageVersion: LanguageMajorVersion) {
       )
   }
 
+  sealed case class UpgradeError(message: String) extends SubmitError {
+    // TODO https://github.com/digital-asset/daml/issues/18616: use a non-Dev based error code
```

Comment:
TODO: need to map to Daml based data structures here (as defined in `Error.daml`)